## Wholesale Customers Data - XGBoost Classifier Report

## 1. Introduction

The goal of this project is to classify wholesale customers using an XGBoost Classifier. This report covers dataset analysis, preprocessing, model building, evaluation, and conclusions.

## 2. Libraries Used

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import xgboost as xgb
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix


## 3. Dataset Overview

The dataset is loaded using Pandas:

In [2]:
data = pd.read_csv(r"C:\Users\devad\Downloads\Wholesale customers data.csv")

### Columns in the Dataset:  

- **Channel**: The target variable (1 or 2).  
- **Region**: Geographical region of the customer.  
- **Fresh, Milk, Grocery, Frozen, Detergents_Paper, Delicassen**: Different product categories and their purchase amounts.  

### Relationships:  

- Some features (e.g., **Grocery** and **Detergents_Paper**) are highly correlated, indicating a potential relationship between purchasing behaviors.  


## 4. Data Analysis

### Basic Info and Statistics

In [3]:
data.info()
data.describe()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 440 entries, 0 to 439
Data columns (total 8 columns):
 #   Column            Non-Null Count  Dtype
---  ------            --------------  -----
 0   Channel           440 non-null    int64
 1   Region            440 non-null    int64
 2   Fresh             440 non-null    int64
 3   Milk              440 non-null    int64
 4   Grocery           440 non-null    int64
 5   Frozen            440 non-null    int64
 6   Detergents_Paper  440 non-null    int64
 7   Delicassen        440 non-null    int64
dtypes: int64(8)
memory usage: 27.6 KB


Channel      Region          Fresh          Milk       Grocery  \
count  440.000000  440.000000     440.000000    440.000000    440.000000   
mean     1.322727    2.543182   12000.297727   5796.265909   7951.277273   
std      0.468052    0.774272   12647.328865   7380.377175   9503.162829   
min      1.000000    1.000000       3.000000     55.000000      3.000000   
25%      1.000000    2.000000    3127.750000   1533.000000   2153.000000   
50%      1.000000    3.000000    8504.000000   3627.000000   4755.500000   
75%      2.000000    3.000000   16933.750000   7190.250000  10655.750000   
max      2.000000    3.000000  112151.000000  73498.000000  92780.000000   

             Frozen  Detergents_Paper    Delicassen  
count    440.000000        440.000000    440.000000  
mean    3071.931818       2881.493182   1524.870455  
std     4854.673333       4767.854448   2820.105937  
min       25.000000          3.000000      3.000000  
25%      742.250000        256.750000    408.250000  
50%     1526.000000        816.500000    965.500000  
75%     3554.250000       3922.000000   1820.250000  
max    60869.000000      40827.000000  47943.000000

- **No null values** are found in the dataset.
- The dataset contains **440 rows and 8 columns**.

## Checking for Null Values

In [4]:
data.isnull().sum()


Channel             0
Region              0
Fresh               0
Milk                0
Grocery             0
Frozen              0
Detergents_Paper    0
Delicassen          0
dtype: int64

## 5. Data Preprocessing


### Feature and Target Selection

In [5]:
x = data.drop(['Channel'], axis=1)
y = data['Channel']


- **X (features)**: All columns except `Channel`

- **Y (target variable)**: `Channel`

### Splitting the Data

In [6]:
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.3, random_state=0)

- 70% training data, 30% testing data.

## 6. Model Building

### Initializing and Training the XGBoost Classifier

In [ ]:
from xgboost import XGBClassifier

params = {
    'objective': 'binary:logistic',
    'max_depth': 4,
    'alpha': 10,
    'learning_rate': 0.1,
    'n_estimators': 100
}

xgb_clf = XGBClassifier(**params)
xgb_clf.fit(x_train, y_train)


### Predictions

In [ ]:
y_pred = xgb_clf.predict(x_test)

## 7. Model Evaluation

### Accuracy Score

In [ ]:
accuracy = accuracy_score(y_test, y_pred) * 100
print("The Accuracy Score:", accuracy)


Accuracy: 87.88%